ui/pages/routes/routes.component.js

```diff
@@ -87,10 +85,12 @@ export default class Routes extends Component {
     isMouseUser: PropTypes.bool,
     setMouseUserState: PropTypes.func,
     providerId: PropTypes.string,
+    providerType: PropTypes.string,
     autoLockTimeLimit: PropTypes.number,
     pageChanged: PropTypes.func.isRequired,
     prepareToLeaveSwaps: PropTypes.func,
-    browserEnvironment: PropTypes.object,
```

Comment:
You are right. We could use `React.memo` with the previous props (e.g. `browserEnvironment`) and it would minimize re-renders as well. I didn't notice this solution before.

However, I still prefer the above solution for a couple reasons. Firstly, it makes the props data more explicit, which will make the component easier to read and follow. Secondly, it will be slightly faster than `React.memo` because there won't be the overhead cost of memoizing the component and calling `isEqual` every time there is a data change.